### PR TITLE
Textcontent

### PR DESCRIPTION
**Pull request by azakus**
Add spec compliant `getTextContent`, `setTextContent`, and `normalize` functions

### CODE REVIEW

**Comment by azakus** [comment]
One caveat, normalized text nodes are always new nodes. Spec wants them to be mutated, but new nodes is far simpler.

**Comment by azakus** [comment]
Added support for normalize on document and document fragment nodes.

File: dom5.js [hunk 1]
```diff
@@ -73,23 +73,82 @@ function hasClass(name) {
...
+    tn.parentNode = parent;
+    parent.childNodes.splice(start, 0, tn);
+  }
+}
+
 /**
  * Normalize the text inside an element
  *
- * Equivalent to `element.textContent` in the browser
+ * Equivalent to `element.normalize()` in the browser
```
Comment (garlicnation) [inline dom5.js]: Link to https://developer.mozilla.org/en-US/docs/Web/API/Node/normalize?

**Comment by garlicnation** [comment]
Is it still true that normalized nodes are new vs mutated?

**Comment by garlicnation** [comment]
LGTM.

**Comment by azakus** [comment]
Yes, normalized nodes are new instead of mutated in this implementation. I could probably try to mutate the "lead" text node instead, but it would be a bit more complicated.

File: dom5.js (hunk 1 above)
Comment (azakus) [inline dom5.js]: Done

**Pull request merged**